target_disk_unix.go

```diff
@@ -6,11 +6,21 @@
 package extract
 
 import (
+	"io/fs"
+	"os"
 	"time"
 
 	"golang.org/x/sys/unix"
 )
 
+// Chown changes the numeric uid and gid of the named file.
+func (d *TargetDisk) Chown(name string, uid, gid int) error {
+	if os.Geteuid() != 0 {
+		return &fs.PathError{Op: "chown", Path: name, Err: os.ErrPermission}
+	}
+	return os.Lchown(name, uid, gid)
```

Comment:
I wonder if we should just let this fail, then check if the failure is a permissions error, and if so maybe wrap it with a hint "only root users can chown"? Just in case there is some unix OS where non-root users can chown, I'm not sure if there is or not but there are [a _lot_](https://github.com/golang/go/blob/3bd08b97921826c1b0a5fbf0789f4b49d7619977/src/internal/syslist/syslist.go#L38-L54) of unix OSs so maybe it's possible?

Reply:
Mhm .. I get what you say

> _... only root users can chown ..._

I am not certain about that statement, bc/ I am not 100% confident about that. So, taking your recommendation and wrapping the error sounds reasonable to me 👌

Reply:
Addressed in https://github.com/hashicorp/go-extract/commit/668eb6b4cadd39a53728752a685d69ce2a8f5543